### PR TITLE
Session refactoring

### PR DESCRIPTION
**Pull request by mtraneis**
Refactored the code to use the SessionData pattern. 
- Eliminated HistoricalMessageRepository and its tests
- Store PreviousMessage instance in SessionData

In order to test ACS request resend scenarios, which require to have an existing Previous request/response we need to make two calls, one to set up the first request, the second to make use of the first reequest's response, which is stored in SessionData's Previous Message. Modified BaseTest to have a callService method that reuses the socket to be able to call multiple times. This method takes multiple sip request messages, but only with one validation handler for now. The method loops through the messages and call the SIP service one by one. 

- also added a couple of IntelliJ entries to git ignore file.

### CODE REVIEW

File: src/test/java/org/folio/edge/sip2/api/support/BaseTest.java [hunk 1]
```diff
@@ -4,6 +4,7 @@
 import static org.mockito.Mockito.any;
 import static org.mockito.Mockito.when;
 
+import io.vertx.core.AsyncResult;
```
Comment (mreno-EBSCO) [inline src/test/java/org/folio/edge/sip2/api/support/BaseTest.java]: How is this import used? I don't see it in the new code.

File: src/test/java/org/folio/edge/sip2/api/support/BaseTest.java [hunk 2]
```diff
@@ -85,6 +86,48 @@ void shutdown(Vertx vertx, TestInfo testInfo) {
     log.info("Finished: {}", testInfo.getDisplayName());
   }
 
+  /**
+   * Calls the service.
+   * @param ncipMessage the sip message to send.
```
Comment (mreno-EBSCO) [inline src/test/java/org/folio/edge/sip2/api/support/BaseTest.java]: Just out of curiosity, why `ncipMessage` and not `sipMessage` here? I've seen that in some other test code and thought, "Hmmm...".

Comment (mtraneis) [reply src/test/java/org/folio/edge/sip2/api/support/BaseTest.java]: typos :) I'll fixed them.

File: src/test/java/org/folio/edge/sip2/api/support/BaseTest.java (hunk 1 above)
Comment (mtraneis) [reply src/test/java/org/folio/edge/sip2/api/support/BaseTest.java]: Weird, I don't have it in my local copy. Thought I had pushed everything up already.

**Pull request merged**